### ISSUE TITLE
Help Command Omitting Commands

### ISSUE THREAD

**Issue by tophattwaffle** [tophattwaffle/BotHATTwaffle#16]
>Help command will omit commands that have [RequireContext(ContextType.Guild)] attribute on them when a check seeing if some precondition is set. @MarkKoz can you review this and see if the check is actually needed? Currently running without it.

See Help.cs at line 154 - 157

**Comment by MarkKoz** [comment]
It's not a bug. `CheckPreconditionsAsync` returns false if any precondition fails in the context in which the command was invoked. As `RequireContext` is a precondition, if it is not met, the function will return false. `RequireContext(ContextType.Guild)` is not met when the `help` command is invoked in a DM context, therefore any commands that are disallowed in DMs are omitted from the list.

That check was there before I touched the code and I left it in. It just had no effect before because preconditions were not used anywhere. It's not needed for the help command to work. Remove it if you do not want it. This was a concern of mine when I first encountered this line, sorry for not bringing it up. Anyway, I learned that it'd be too difficult to only check certain types of preconditions e.g. permissions & roles but not contexts.
